### PR TITLE
feat: Add ability to test hot-reload scenario

### PR DESCRIPTION
**Pull request by dr1rrb**
fix: https://github.com/unoplatform/uno.ui.runtimetests.engine/issues/87
fix: https://github.com/unoplatform/uno.ui.runtimetests.engine/issues/102

## Feature
Add ability to test hot-reload scenario

## What is the current behavior?
* No way to test hot-reload
* Missing lot of helpers to test applications

## What is the new behavior?
* Add support for auto-start of external application
* Import helpers from uno and other projects

## PR Checklist
- [x] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [x] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [x] Associated with an issue (GitHub or internal)

### CODE REVIEW

File: src/Uno.UI.RuntimeTests.Engine.Library/Engine/ExternalRunner/RuntimeTestsSourceProjectAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Uno.UI.RuntimeTests.Engine;
+
+[AttributeUsage(AttributeTargets.Assembly, AllowMultiple = false)]
+public sealed class RuntimeTestsSourceProjectAttribute : Attribute
+{
+	// Note: This is somehow a duplicate of the Uno.UI.RemoteControl.ProjectConfigurationAttribute but it allows us to have the attribute even in release ==>> TODO: which is needed anyway ???
```
Comment (dr1rrb) [inline src/Uno.UI.RuntimeTests.Engine.Library/Engine/ExternalRunner/RuntimeTestsSourceProjectAttribute.cs:17]: Note: No longer a TODO, will remove in my coming PR (with update of Uno)

File: src/Uno.UI.RuntimeTests.Engine.Library/Library/Helpers/HotReloadHelper.cs [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+
+namespace Uno.UI.RemoteControl.HotReload.Messages
+{
+	public partial class UpdateFile
+	{
+		public string FilePath { get; set; } = string.Empty;
+		public string OldText { get; set; } = string.Empty;
+		public string NewText { get; set; } = string.Empty;
+	}
+}
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests.Engine.Library/Library/Helpers/HotReloadHelper.cs:57]: Move to separate files

**Review by jeromelaban — APPROVED** [review]

File: src/TestApp/uwp/Uno.UI.RuntimeTests.Engine.UWP/Uno.UI.RuntimeTests.Engine.Uwp.csproj [hunk 3]
```diff
@@ -37,6 +37,7 @@
     <MinimumVisualStudioVersion>14</MinimumVisualStudioVersion>
     <FileAlignment>512</FileAlignment>
     <ProjectTypeGuids>{A5A43C5B-DE2A-4C0C-9213-0A381AF9435A};{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}</ProjectTypeGuids>
+	<DefineConstants>$(DefineConstants);UNO_RUNTIMETESTS_DISABLE_HOTRELOAD</DefineConstants>
```
[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/TestApp/uwp/Uno.UI.RuntimeTests.Engine.UWP/Uno.UI.RuntimeTests.Engine.Uwp.csproj]: This is only the sample app, but it's no longer needed, remove it

File: src/TestApp/winui/Uno.UI.RuntimeTests.Engine.Windows/Uno.UI.RuntimeTests.Engine.Windows.csproj [hunk 4]
```diff
@@ -28,7 +27,7 @@
 		<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.1.3" />
 		<PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22000.196" />
 		<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" />
-		<PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
+		<PackageReference Include="Newtonsoft.Json" Version="13.0.2" />
```
[the remark this answers is not in the document]
Comment (dr1rrb) [reply src/TestApp/winui/Uno.UI.RuntimeTests.Engine.Windows/Uno.UI.RuntimeTests.Engine.Windows.csproj]: It's still being use to save the preferences of the test control ... which can definitively be moved to System.Text, even if not really the purpose of this PR, I removed it as part of the dependencies updates.

**Pull request merged**